ui/src/task-view.tsx

```diff
@@ -119,18 +119,28 @@ export const TaskView = ({ taskId }: { taskId: string }) => {
             >
               +
             </button>
-            <div>
+            <div class="adding-labels">
               {isEditingLabels ? (
                 <>
-                <button onClick={showLabelsPopup}>Edit Labels</button>
-                {allLabels.map((label) => {
-                  return(
-                    <button class="labels-in-box" style={{ background: label.color }}>
-                  {label.name}
-                </button>
-                  )
-                })}
-
+                  <button class="edit-labels-button" onClick={showLabelsPopup}>
+                    Edit Labels
+                  </button>
+                  <input
+                    type="text"
+                    placeholder="search for labels"
+                    onInput={(e) => {
+                      setLabelSearch(e.currentTarget.value)
+                    }}
+                  />
+                  {allLabels.map((label) => {
+                    return label.name.startsWith(labelSearch) ? (
```

Comment:
Can you use [`.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) so that the search does not have to be at the beginning of the string?
```suggestion
                    return label.name.includes(labelSearch) ? (
```